examples/shv-nxboot-updater/shv_nxboot_updater.c

```diff
@@ -0,0 +1,424 @@
+
+static int shv_nxboot_opener(shv_file_node_t *item);
+static int shv_root_device_type(shv_con_ctx_t * shv_ctx, shv_node_t *item,
+                                int rid);
+static int shv_dotapp_vmajor(shv_con_ctx_t *shv_ctx, shv_node_t *item,
+                             int rid);
+static int shv_dotapp_vminor(shv_con_ctx_t *shv_ctx, shv_node_t *item,
+                             int rid);
+static int shv_dotapp_name(shv_con_ctx_t *shv_ctx, shv_node_t *item,
+                           int rid);
+static int shv_dotapp_ping(shv_con_ctx_t *shv_ctx, shv_node_t *item,
+                           int rid);
+static int shv_fwstable_confirm(shv_con_ctx_t *shv_ctx, shv_node_t *item,
+                                int rid);
+static void quit_handler(int signum);
+static void print_help(char *name);
+
+static shv_node_t *shv_tree_create(void);
+static void attention_cb(shv_con_ctx_t *shv_ctx,
+                         enum shv_attention_reason r);
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/* An execution barrier */
+
+static sem_t running;
+
+/* ------------------------- ROOT METHODS --------------------------------- */
+
+static const shv_method_des_t shv_dev_root_dmap_item_device_type =
+{
+  .name = "deviceType",
+  .method = shv_root_device_type
+};
+
+static const shv_method_des_t * const shv_dev_root_dmap_items[] =
+{
+  &shv_dev_root_dmap_item_device_type,
+  &shv_dmap_item_dir,
+  &shv_dmap_item_ls,
+};
+
+static const shv_dmap_t shv_dev_root_dmap =
+  SHV_CREATE_NODE_DMAP(root, shv_dev_root_dmap_items);
+
+/* ------------------------- .app METHODS ---------------------------- */
+
+static const shv_method_des_t shv_dev_dotapp_dmap_item_vmajor =
+{
+  .name = "shvVersionMajor",
+  .method = shv_dotapp_vmajor
+};
+
+static const shv_method_des_t shv_dev_dotapp_dmap_item_vminor =
+{
+  .name = "shvVersionMinor",
+  .method = shv_dotapp_vminor
+};
+
+static const shv_method_des_t shv_dev_dotapp_dmap_item_name =
+{
+  .name = "name",
+  .method = shv_dotapp_name
+};
+
+static const shv_method_des_t shv_dev_dotapp_dmap_item_ping =
+{
+  .name = "ping",
+  .method = shv_dotapp_ping
+};
+
+static const shv_method_des_t * const shv_dev_dotapp_dmap_items[] =
+{
+  &shv_dmap_item_dir,
+  &shv_dmap_item_ls,
+  &shv_dev_dotapp_dmap_item_name,
+  &shv_dev_dotapp_dmap_item_ping,
+  &shv_dev_dotapp_dmap_item_vmajor,
+  &shv_dev_dotapp_dmap_item_vminor,
+};
+
+static const shv_dmap_t shv_dev_dotapp_dmap =
+  SHV_CREATE_NODE_DMAP(dotapp, shv_dev_dotapp_dmap_items);
+
+/* ------------------------- fwstable METHODS ---------------------------- */
+
+static const shv_method_des_t shv_dev_fwstable_dmap_item_confirm =
+{
+  .name = "confirm",
+  .method = shv_fwstable_confirm
+};
+
+static const shv_method_des_t * const shv_dev_fwstable_dmap_items[] =
```

Comment:
there should also be `fwStable:get` method that returns `true` or `false` based on fw status.